project-docs/javascript-linter-spec.md

```diff
@@ -0,0 +1,163 @@
+- [javascript-error](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-error.yaml)
+- [javascript-language-feature](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-language-feature.yaml)
+
+## JavaScript recipe ingredients
+
+In total these recipes use the following ingredients:
+
+prose ingredients:
+
+- prose.short_description
+- prose.description?
+- prose.see_also
+- prose.syntax
+- prose.message
+- prose.error_type
+- prose.what_went_wrong
+- prose.\*
+
+data ingredients:
+
+- data.interactive_example?
+- data.class_constructor
+- data.constructor_properties?
+- data.static_properties?
+- data.static_methods?
+- data.instance_properties?
+- data.instance_methods?
+- data.examples
+- data.specifications
+- data.browser_compatibility
+
+All ingredients except two are demarcated in the page using an `<h2>` heading. The exceptions are `prose.short_description` and `data.interactive_example`.
+
+### Prose ingredients
+
+With three exceptions, all prose ingredients impose the same requirement on the page: it must have an `<h2>` section whose ID attribute maps to the named section. The mapping is simply that `prose.` is removed, and the first letter is capitalized:
+
+```
+prose.syntax      ->    Syntax
+prose.see_also    ->    See_also
+```
+
+The exceptions are `prose.short_description`, `prose.description?`, and `prose.*`.
+
+#### prose.short_description
+
+This ingredient isn't demarcated by an `<h2>` at all. To satisfy the ingredient, a page must:
+
+- contain a `<p>` element
+- that is not a note or a warning
+- before the interactive example or the first `<h2>`, whichever comes first
+
+#### prose.description?
+
+This is an optional prose ingredient. To satisfy the ingredient, if the page contains an `h2#Description`, then it must appear in the place specified by the recipe.
+
+#### prose.\*
+
+This ingredient enables writers to include extra sections, but ensuring that they always appear in the same place in the page. To satisfy this ingredient: if the page contains any `<h2>` headings that are not expected by the recipe, then they must appear at the point in the page that the `prose.*` ingredient is listed in the recipe.
+
+### Data ingredients
+
+Data ingredients typically impose more detailed requirements on pages than prose ingredients.
+
+#### data.interactive_example?
+
+This is an optional data ingredient. To satisfy the ingredient, if the page contains a call to the `{{EmbedInteractiveExample}}` macro, then the call must appear in the place specified by the recipe.
+
+#### data.class_constructor
+
+To satisfy this ingredient, a page must meet the following requirements:
+
+- A section demarcated by `H2#Constructor` that contains only a `<dl>` element.
+- The `<dl>` must contain a single `<dt><dd>` pair.
+- The `<dt>` must contain either:
+  - only a single `<code>` element, that contains only a single `<a>` element
+  - only a call to the `{{jsxref}}` macro.
```

Comment:
If we are getting this specific, I also tried to always have this format in the `<dd>` element: 

`"Creates a new <code>NameOfTheObject</code> object."`

Maybe I'm going too far now, though :)

Reply:
I have added this: https://github.com/mdn/stumptown-content/pull/391/commits/725dba8d4ad88ec9f17be4c3c93052eda5d00a98 . I'm a bit worried it will be hard to test, especially getting `NameOfTheObject`, but let's revise it later if it looks too hard.